pkg/controller/v1beta1/trainedmodel/shard/shard_manager.go

```diff
@@ -0,0 +1,21 @@
+package shard
+
+import (
+	v1beta1api "github.com/kubeflow/kfserving/pkg/apis/serving/v1beta1"
+)
+
+type ShardingStrategy string
+
+const (
+	Memory ShardingStrategy = "Memory"
```

Comment:
I'd model this as multiple structs implementing an interface, rather than one mega class that handles all strategies.

Reply:
Added an interface and created a dummy MemoryStrategy
```
type Strategy interface {
	GetOrAssignShard(trainedModel *v1beta1api.TrainedModel) int
}

//TODO MemoryShardingStrategy will be implemented in another PR
type MemoryShardingStrategy struct {

}

// Return a TrainedModel's shardId
func (v *MemoryShardingStrategy) GetOrAssignShard(trainedModel *v1beta1api.TrainedModel) int {
	//TODO to be implemented in another PR
	//Currently each InferenceService only has one sharding with id=0
	return 0
}
```